src/abstractdataframe/abstractdataframe.jl

```diff
@@ -75,40 +74,8 @@ _names(df::AbstractDataFrame) = _names(index(df))
 
 Compat.hasproperty(df::AbstractDataFrame, s::Symbol) = haskey(index(df), s)
 
-"""
-Set column names
-
-
-```julia
-names!(df::AbstractDataFrame, vals)
-```
-
-**Arguments**
-
-* `df` : the AbstractDataFrame
-* `vals` : column names, normally a Vector{Symbol} the same length as
-  the number of columns in `df`
-* `makeunique` : if `false` (the default), an error will be raised
-  if duplicate names are found; if `true`, duplicate names will be suffixed
-  with `_i` (`i` starting at 1 for the first duplicate).
-
-**Result**
-
-* `::AbstractDataFrame` : the updated result
-
-
-**Examples**
-
-```julia
-df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
-names!(df, [:a, :b, :c])
-names!(df, [:a, :b, :a])  # throws ArgumentError
-names!(df, [:a, :b, :a], makeunique=true)  # renames second :a to :a_1
-```
-
-"""
-function names!(df::AbstractDataFrame, vals; makeunique::Bool=false)
-    names!(index(df), vals, makeunique=makeunique)
+function rename!(df::AbstractDataFrame, vals::Vector{Symbol}; makeunique::Bool=false)
```

Comment:
How about allowing an `AbstractVector{Symbol}` everywhere?

Reply:
I will change this. I left `Vector{Symbol}` as this is what was originally there and I thought having another collection of `Symbol`s is unlikely so I left it as is. But it is OK to change this.